realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.java

```diff
@@ -226,6 +242,10 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                 }
             }
 
+            if (!hasDefaultConstructor) {
+                error("A no arg default public constructor must also be available if a custom constructor is present");
```

Comment:
Please consider rephrasing; A default public constructor with no argument must be declared if a custom constructor is declared.